apps/aecore/lib/aecore/tx/signed_tx.ex

```diff
@@ -56,21 +75,42 @@ defmodule Aecore.Tx.SignedTx do
      - priv_key: The priv key to sign with
 
   """
-  @spec sign_tx(DataTx.t(), binary()) :: {:ok, SignedTx.t()}
-  def sign_tx(%DataTx{} = tx, priv_key) when byte_size(priv_key) == 32 do
-    signature = Signing.sign(Serialization.pack_binary(tx), priv_key)
 
-    if byte_size(signature) <= get_sign_max_size() do
-      {:ok, %SignedTx{data: tx, signature: signature}}
+  @spec sign_tx(DataTx.t() | SignedTx.t(), binary(), binary()) ::
+          {:ok, SignedTx.t()} | {:error, binary()}
+  def sign_tx(%DataTx{} = tx, pub_key, priv_key) do
+    signatures =
+      for _ <- DataTx.senders(tx) do
```

Comment:
Why do you fill the signatures with `nil` values?

Reply:
Because signatures can be added in any order and this makes ordering signatures easier.